pkg/recommender/engine.go

```diff
@@ -71,6 +65,90 @@ func (e *Engine) RecommendCluster(provider string, service string, region string
 		}
 	}
 
+	var cheapestMaster []NodePool
+	var wg sync.WaitGroup
+	errorChan := make(chan error, 1)
+	defer close(errorChan)
+	nodePoolChan := make(chan []NodePool, 1)
+	defer close(nodePoolChan)
+
+	if (service == "pke" || service == "ack") && layoutDesc == nil {
+		wg.Add(1)
+		go func(prv, service string, req ClusterRecommendationReq, allProducts []VirtualMachine, wg *sync.WaitGroup) {
```

Comment:
Please extract this inline function into an unexported method of the engine. Also is there any reason to execute this call in a new goroutine? As far as i can see this can be a simple synchronous call (so chan-s and the sync usage can be removed)

Another thought: please consider recommending masters a  new method in the ClusterRecommender interface: this way all the custom master recommendation logic -which will probably evolve further -could be separated